src/main/java/com/bobocode/hoverla/bring/context/BeanValidator.java

```diff
@@ -1,18 +1,113 @@
 package com.bobocode.hoverla.bring.context;
 
-import com.bobocode.hoverla.bring.exception.NoUniqueBeanException;
+import com.bobocode.hoverla.bring.exception.BeanValidationException;
+import lombok.extern.slf4j.Slf4j;
 
 import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
 
 /**
- * Validates beans list to avoid duplicates or other conflicts
+ * Validates beans list to avoid such cases: duplicate names, bean name is null, empty, or contains spaces, bean type
+ * is not suitable for the required type, circular dependency, application context doesn't contain the required bean
  */
+@Slf4j
 public class BeanValidator {
 
     /**
-     * @throws NoUniqueBeanException if there's bean duplicate found
+     * @param beanDefinitions all bean definitions of application context
+     * @throws NullPointerException    if beanDefinitions is null
+     * @throws BeanValidationException if there are conflicts: duplicate names, bean name is null, empty,
+     *                                 or contains spaces, bean type is not suitable for the required type,
+     *                                 circular dependency, application context doesn't contain the required bean
      */
     public void validate(List<BeanDefinition> beanDefinitions) {
+        log.debug("Bean validation started");
+        Objects.requireNonNull(beanDefinitions, "Bean definition list is null");
 
+        beanDefinitions.forEach(beanDefinition -> {
+            checkBeanName(beanDefinition);
+            checkDependencies(beanDefinition, beanDefinitions);
+        });
+        Map<String, Long> duplicateNames = beanDefinitions.stream()
+                .collect(Collectors.groupingBy(BeanDefinition::name, Collectors.counting()))
+                .entrySet()
+                .stream()
+                .filter(v -> v.getValue() > 1)
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        if (!duplicateNames.isEmpty()) {
+            throw new BeanValidationException("Context contains beans with duplicate names: " + duplicateNames.keySet());
+        }
+    }
+
+    private void checkBeanName(BeanDefinition beanDefinition) {
+        log.trace("Checking name for bean definition: {} - {}", beanDefinition.name(), beanDefinition.type());
+        String name = beanDefinition.name();
+        if (beanDefinition.name() == null || name.isEmpty() || name.contains(" ")) {
```

Comment:
better use `isBlank()` method. It checks whether the string contains only whitespaces
`if (name == null || name.isBlank())`

Reply:
name.isBlank throws NPE if name == null. I will change `beanDefinition.name() == null || name.isEmpty()` on `Strings.isNullOrEmpty(name)`. Regarding whitespace, need checking on containing spaces inside a string.